src/java.base/share/classes/javax/crypto/KDFSpi.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (c) 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package javax.crypto;
+
+import javax.crypto.spec.KDFParameterSpec;
+import java.security.InvalidAlgorithmParameterException;
+import java.security.spec.AlgorithmParameterSpec;
+import java.security.spec.InvalidParameterSpecException;
+
+/**
+ * This class defines the <i>Service Provider Interface</i> (<b>SPI</b>) for the
+ * <code>KeyDerivation</code> class.
+ * <p>
+ * All the abstract methods in this class must be implemented by each cryptographic service provider
+ * who wishes to supply the implementation of a particular key derivation algorithm.
+ */
+public abstract class KDFSpi {
+
+    /**
+     * provides access to the KDF alg params for implementers of the Spi
+     */
+    protected final AlgorithmParameterSpec algorithmParameterSpec;
```

Comment:
Putting the parameter object here means every time a subclass wants to use it it needs to cast it to a child type. If it's not used inside this class, I suggest each sub class maintaining its own.

Reply:
Yes, I don't think should be exposed as a protected field since it is passed to the constructor and subclasses can store it as a private variable.